### DISCUSSION TITLE
mitmproxy 9.0.1 connection to https://www.legislation.gov.au/ results in sslv3 alert handshake failure

### DISCUSSION THREAD

**Discussion by morb-au-public** [mitmproxy/mitmproxy#6062] [General]
Hi everyone,

Can someone else check and let me know if mitmproxy can connect to https://www.legislation.gov.au/ ?  

I am getting:

`OpenSSL Error([('SSL routines', '', 'sslv3 alert handshake failure')])
`

Background:

Wireshark is showing a response with the handshake_failure(40).

I'm new to mitmproxy and just want to see if this scenario is expected/supported.

In contrast, I can get mitmproxy working on a few other sites, like http://mitm.it/ and https://www.legislation.qld.gov.au/

My environment is Ubuntu on WSL2 on Windows 11:

```
$ mitmproxy --version
Mitmproxy: 9.0.1
Python:    3.10.6
OpenSSL:   OpenSSL 3.0.7 1 Nov 2022
Platform:  Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
```

I've also tried on the native Windows 11 host as well, resulting in with the same handshake_failure(40) from https://www.legislation.gov.au/:

```
mitmproxy --version
Mitmproxy: 9.0.1 binary
Python:    3.11.0
OpenSSL:   OpenSSL 3.0.7 1 Nov 2022
Platform:  Windows-10-10.0.22621-SP0
```

**Comment by sujaldev** [comment]
This doesn't seem like an issue with mitmproxy.

**Reply by morb-au-public** [reply]
Sujal,

So, to confirm, https://www.legislation.gov.au/ works for you through mitmproxy?

Brendan

**Reply by sujaldev** [reply]
No, I meant that the handshake failure exists even without using mitmproxy (at least with curl, it does work in a web browser when not using mitmproxy). Try running `curl -v https://www.legislation.gov.au`, the error still exists. This is probably an issue with openssl on the server, but I'm not sure.
